src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java

```diff
@@ -58,36 +64,47 @@
 
 	private ZMQ.Socket subscriber;
 	private ZMQ.Context context;
+	private ObjectMapper mapper = new ObjectMapper();
 
 	{
 		context = ZMQ.context(1);
 	}
 
 	public void receive() {
 		getSubscriber();
-		String exportString = null;
-		byte[] exportBytes = null;
-		Event event;
+		JsonNode node;
+		ZMsg zmsg;
+		ZFrame[] parts;
 		logger.info("Watching for new exported Event messages...");
 		try {
 			while (!Thread.currentThread().isInterrupted()) {
-				if (exportClient) {
-					if (serializedJava) { // supporting legacy Java export distro that shipped Serialized eventString
-						exportString = subscriber.recvStr();
-						event = toEvent(exportString);
-					} else { // supporting new Go export distro that ships JSON
-						exportBytes = subscriber.recv();
-						event = toEvent(exportBytes);
+				zmsg = ZMsg.recvMsg(subscriber);
+				parts = new ZFrame[zmsg.size()];
+				zmsg.toArray(parts);
+				logger.debug("Message has " + parts.length + " parts.");
+
+				if (parts.length < 2) {// if the message is not a multi-part message as delivered by core data
```

Comment:
Actually the non multi-part messages will be coming from Export Distro. Core Data is using the new Message Bus and send the data with the "events" topic. So the comment just needs `core data` changed to `export distro`

Reply:
updated accordingly